src/Shared/Infrastructure/Provider/TranslationsCatalogProvider.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Shared\Infrastructure\Provider;
+
+use App\PullRequest\Domain\Aggregate\PullRequest\PullRequest;
+use Psr\Cache\InvalidArgumentException;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class TranslationsCatalogProvider implements TranslationsCatalogInterface
+{
+    private const TRANSLATIONS_CATALOG_URL = 'https://raw.githubusercontent.com/PrestaShop/TranslationFiles/master/%s/%s/%s.%s.xlf';
+    private const CACHE_TTL = 3600;
+
+    public function __construct(
+        private readonly HttpClientInterface $httpClient,
+        private readonly CacheInterface $cache
+    ) {
+    }
+
+    /**
+     * Retrieve the PrestaShop version catalog based on the branch name.
+     */
+    public function getCatalogVersionByPullRequest(PullRequest $pullRequest): int
+    {
+        $PSVersion = 9;
+
+        if ('PrestaShop' === $pullRequest->getId()->repositoryName && str_starts_with($pullRequest->getTargetBranch(), '8')) {
```

Comment:
Will this code works with PR targeting branch `1.7.8.x` ?

Reply:
I don't really think that theses `1.7.x` branches are necessary. Maybe I'm wrong?

Reply:
I agree that kanbanbot does not need to verify `1.7.x` branches 👍 we will not add new wordings in `1.7.x` versions

But I'm worried kanbanbot would spam PRs for 1.7.x with invalid messages.

But it's a minot usecase 😄 if it happens we can always fix it later